server-common/src/main/java/org/apache/cassandra/sidecar/common/server/dns/DnsResolver.java

```diff
@@ -26,29 +26,17 @@
  */
 public interface DnsResolver
 {
-    DnsResolver DEFAULT = new DnsResolver()
-    {
-        @Override
-        public String resolve(String hostname) throws UnknownHostException
-        {
-            return InetAddress.getByName(hostname).getHostAddress();
-        }
-
-        @Override
-        public String reverseResolve(String address) throws UnknownHostException
-        {
-            return InetAddress.getByName(address).getHostName();
-        }
-    };
-
     /**
      * Resolves the IP address of the hostname
      *
      * @param hostname the hostname to resolve
      * @return IP address
      * @throws UnknownHostException when the host is not known
      */
-    String resolve(String hostname) throws UnknownHostException;
+    default String resolve(String hostname) throws UnknownHostException
+    {
+        return InetAddress.getByName(hostname).getHostAddress();
+    }
```

Comment:
I would preserve this interface and not add a default, but instead implement `resolve` in the `DnsResolverEnum`
```suggestion
    String resolve(String hostname) throws UnknownHostException;
```